azurerm/resource_arm_api_management.go

```diff
@@ -0,0 +1,773 @@
+	name := apimanagement.SkuType(nameConfig)
+	capacity := int32(config["capacity"].(int))
+
+	sku := &apimanagement.ServiceSkuProperties{
+		Name:     name,
+		Capacity: &capacity,
+	}
+
+	return sku
+}
+
+func flattenApiManagementCertificates(d *schema.ResourceData, props *[]apimanagement.CertificateConfiguration) []interface{} {
+	certificates := make([]interface{}, 0)
+
+	if props != nil {
+		for i, prop := range *props {
+			certificate := make(map[string]interface{}, 0)
+
+			certificate["store_name"] = string(prop.StoreName)
+
+			// certificate password isn't returned, so let's look it up
+			passwKey := fmt.Sprintf("certificate.%d.certificate_password", i)
+			if v, ok := d.GetOk(passwKey); ok {
+				password := v.(string)
+				certificate["certificate_password"] = password
+			}
+
+			// encoded certificate isn't returned, so let's look it up
+			certKey := fmt.Sprintf("certificate.%d.encoded_certificate", i)
+			if v, ok := d.GetOk(certKey); ok {
+				cert := v.(string)
+				certificate["encoded_certificate"] = cert
+			}
+
+			certificates = append(certificates, certificate)
+		}
+	}
+
+	return certificates
+}
+
+func flattenApiManagementAdditionalLocations(props *[]apimanagement.AdditionalLocation) []interface{} {
+	additional_locations := make([]interface{}, 0)
+
+	if props != nil {
+		for _, prop := range *props {
+			additional_location := make(map[string]interface{}, 0)
+
+			if prop.Location != nil {
+				additional_location["location"] = *prop.Location
+			}
+
+			if prop.StaticIps != nil {
+				additional_location["static_ips"] = *prop.StaticIps
+			}
+
+			if prop.GatewayRegionalURL != nil {
+				additional_location["gateway_regional_url"] = *prop.GatewayRegionalURL
+			}
+
+			additional_locations = append(additional_locations, additional_location)
+		}
+	}
+
+	return additional_locations
+}
+
+func flattenApiManagementHostnameConfigurations(d *schema.ResourceData, configs *[]apimanagement.HostnameConfiguration) []interface{} {
+	host_configs := make([]interface{}, 0)
+
+	if configs != nil {
+		for i, config := range *configs {
+			host_config := make(map[string]interface{}, 0)
+
+			if config.Type != "" {
+				host_config["type"] = string(config.Type)
+			}
+
+			if config.HostName != nil {
+				host_config["host_name"] = *config.HostName
+			}
+
+			if config.DefaultSslBinding != nil {
+				host_config["default_ssl_binding"] = *config.DefaultSslBinding
+			}
+
+			if config.NegotiateClientCertificate != nil {
+				host_config["negotiate_client_certificate"] = bool(*config.NegotiateClientCertificate)
+			}
+
+			// certificate password isn't returned, so let's look it up
+			passKey := fmt.Sprintf("hostname_configuration.%d.certificate_password", i)
+			if v, ok := d.GetOk(passKey); ok {
+				password := v.(string)
+				host_config["certificate_password"] = password
+			}
+
+			// encoded certificate isn't returned, so let's look it up
+			certKey := fmt.Sprintf("hostname_configuration.%d.certificate", i)
```

Comment:
You will have to look these up by name, if someone edits this on the portal the index's may no longer match.